storage/engine/gc.go

```diff
@@ -88,7 +92,11 @@ func (gc *GarbageCollector) Filter(keys []proto.EncodedKey, values [][]byte) pro
 	// If there are no non-deleted survivors, return timestamp of first key
 	// to delete all entries.
 	if !survivors {
-		_, ts, _ := MVCCDecodeKey(keys[0])
+		_, ts, _, err := MVCCDecodeKey(keys[0])
+		if err != nil {
+			log.Errorf("unable to decode MVCC key: %q: %v", keys[0], err)
```

Comment:
if you can't decode it, I'm not sure you want to gc everything. Better to leave it in place for eventual inspection - but of course there's some value in reporting this up (potential ReplicaCorruptionError). Didn't we have a crawler queue whose sole purpose it is to touch everything?